modules/javafx.graphics/src/main/java/com/sun/glass/ui/GlassRobot.java

```diff
@@ -0,0 +1,363 @@
+                                          double height, boolean scaleToFit) {
+        if (width <= 0) {
+            throw new IllegalArgumentException("width must be > 0");
+        }
+        if (height <= 0) {
+            throw new IllegalArgumentException("height must be > 0");
+        }
+        Screen primaryScreen = Screen.getPrimary();
+        Objects.requireNonNull(primaryScreen);
+        double outputScaleX = primaryScreen.getOutputScaleX();
+        double outputScaleY = primaryScreen.getOutputScaleY();
+        int data[];
+        int dw, dh;
+        if (outputScaleX == 1.0f && outputScaleY == 1.0f) {
+            // No scaling will be necessary regardless of if "scaleToFit" is set or not.
+            data = new int[(int) (width * height)];
+            getScreenCapture((int) x, (int) y, (int) width, (int) height, data, scaleToFit);
+            dw = (int) width;
+            dh = (int) height;
+        } else {
+            // Compute the absolute pixel bounds that the requested size will fill given
+            // the display's scale.
+            int pminx = (int) Math.floor(x * outputScaleX);
+            int pminy = (int) Math.floor(y * outputScaleY);
+            int pmaxx = (int) Math.ceil((x + width) * outputScaleX);
+            int pmaxy = (int) Math.ceil((y + height) * outputScaleY);
+            int pwidth = pmaxx - pminx;
+            int pheight = pmaxy - pminy;
+            int tmpdata[] = new int[pwidth * pheight];
+            getScreenCapture(pminx, pminy, pwidth, pheight, tmpdata, scaleToFit);
+            dw = pwidth;
+            dh = pheight;
+            if (!scaleToFit) {
+                data = tmpdata;
+            } else {
+                // We must resize the image to fit the requested bounds. This means
+                // resizing the pixel data array which we accomplish using bilinear (?)
+                // interpolation.
+                data = new int[(int) (width * height)];
+                int index = 0;
+                for (int iy = 0; iy < height; iy++) {
+                    double rely = ((y + iy + 0.5f) * outputScaleY) - (pminy + 0.5f);
+                    int irely = (int) Math.floor(rely);
+                    int fracty = (int) ((rely - irely) * 256);
+                    for (int ix = 0; ix < width; ix++) {
+                        double relx = ((x + ix + 0.5f) * outputScaleX) - (pminx + 0.5f);
+                        int irelx = (int) Math.floor(relx);
+                        int fractx = (int) ((relx - irelx) * 256);
+                        data[index++] = interp(tmpdata, irelx, irely, pwidth, pheight, fractx, fracty);
+                    }
+                }
+                dw = (int) width;
+                dh = (int) height;
+            }
+        }
+
+        return convertFromPixels(image, Application.GetApplication().createPixels(dw, dh, IntBuffer.wrap(data)));
+    }
+
+    public static int convertToRobotMouseButton(MouseButton[] buttons) {
```

Comment:
Public method w/o the spec?

Reply:
This is not public API.

Reply:
Ok, but this file looks inconsistent  because some methods has spec and others not.

Reply:
Perhaps this could be good cleanup.

Reply:
I just added them because they were easy to copy from `Robot.java` and I figured some docs is better than no docs :)